ksql-rest-app/src/test/java/io/confluent/ksql/rest/util/ClusterTerminatorTest.java

```diff
@@ -104,31 +106,47 @@ public void shouldCloseTheEngineAfterTerminatingPersistetQueries() throws Except
     verify(ksqlEngine).close();
   }
 
+  @Test
+  public void shouldTerminateQueriesBeforeDeleteingTopics() {
```

Comment:
This test does not test 'TerminateQueriesBeforeDeleteingTopics'. It tests that the code calls `getPersistentQueries` and `listTopicNames` before it deletes any topics. This is completely different.

The test _should_ have `getPersistentQueries` return a query, e.g. `persistentQueryMetadata` , (in fact this could be done in setup). Then verify the query is terminated before deletetopics is called. `listTopicNames` is irrelevant to this test.